lib/stimulus_reflex/utils/sanity_checker.rb

```diff
@@ -101,39 +109,41 @@ def check_new_version_available
     end
   end
 
-  private
-
-  def caching_enabled?
-    Rails.application.config.action_controller.perform_caching
+  def caching_not_enabled?
+    Rails.application.config.action_controller.perform_caching == false
   end
 
-  def not_null_store?
-    Rails.application.config.cache_store != :null_store
+  def using_null_store?
+    Rails.application.config.cache_store == :null_store
   end
 
   def default_url_config_set?
-    Rails.application.config.action_controller.default_url_options
+    if defined?(ActionMailer)
+      Rails.application.config.action_controller.default_url_options && Rails.application.config.action_mailer.default_url_options
+    else
+      Rails.application.config.action_controller.default_url_options
+    end
```

Comment:
This will never fire.

The reason is that `default_url_options` returns `nil` but above you check:

`default_url_config_set? == false`

These should probably be blank checks to catch empty hashes and nil. but still return a boolean.

```suggestion
    if defined?(ActionMailer)
      Rails.application.config.action_controller.default_url_options.blank? && Rails.application.config.action_mailer.default_url_options.blank?
    else
      Rails.application.config.action_controller.default_url_options.blank?
    end
```